auxx/src/Lang/DisplayError.hs

```diff
@@ -0,0 +1,141 @@
+module Lang.DisplayError
+    ( ppArgumentError
+    , ppEvalError
+    , ppTypeError
+    , ppTypeName
+    , ppParseError
+    , ppProcError
+    , renderAuxxDoc
+    ) where
+
+import           Universum hiding (empty, (<$>))
+
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Set as S
+import qualified Data.Text.Lazy as T
+import qualified Text.PrettyPrint.ANSI.Leijen
+
+
+import           Control.Lens (matching)
+import           Data.Loc (Span, loc, locColumn, locLine, spanEnd, spanFromTo, spanStart, toNat)
+import           Data.Loc.Span (joinAsc)
+import           Data.Text (unpack)
+import           Data.Text.Buildable (build)
+import           Data.Text.Lazy.Builder (toLazyText)
+import           Text.Earley (Report (..))
+import           Text.PrettyPrint.ANSI.Leijen (Doc, bold, char, comma, empty, hcat, indent,
+                                               punctuate, squotes, vcat, yellow, red, (<$>), (<+>))
+
+import           Lang.Argument (ArgumentError (..), ProcError (..), TypeError (..), TypeName (..),
+                                isEmptyArgumentError)
+import           Lang.Interpreter (EvalError (..))
+import           Lang.Lexer (_TokenUnknown)
+import           Lang.Name (Name)
+import           Lang.Parser (ParseError (..))
+
+
+highlight :: Doc -> Doc
+highlight = bold . yellow
+
+text :: Text -> Doc
+text = Text.PrettyPrint.ANSI.Leijen.text . unpack
+
+nameToDoc :: Name -> Doc
+nameToDoc = Text.PrettyPrint.ANSI.Leijen.text . T.unpack . toLazyText . build
+
+ppTypeName :: TypeName -> Doc
+ppTypeName (TypeName name)         = text name
+ppTypeName (TypeNameEither tn tn') = ppTypeName tn <+> char '|' <+> ppTypeName tn'
+
+ppTypeError :: TypeError -> Doc
+ppTypeError TypeError{..} =
+        "Couldn't match expected type" <+> (highlight $ ppTypeName teExpectedType)
+    <+> "with actual value"            <+> (highlight $ show teActualValue) -- TODO: CSL-1814
+    <>  "!"
+
+ppArgumentError :: ArgumentError -> Doc
+ppArgumentError ae@ArgumentError{..} =
+    if isEmptyArgumentError ae
+    then empty
+    else vcat $ missingKeysDoc <> irrelevantKeysDoc <> irrelevantPosDoc
+  where
+    setToDoc = hcat . punctuate comma . map (highlight . nameToDoc) . S.toList
+    missingKeysDoc =
+        if S.null aeMissingKeys
```

Comment:
Btw, `null` and `toList` from `Universum` are polymorphic, but I don't mind if you use `S.{toList, null}`.